docs/testing-styleguide.MD

```diff
@@ -0,0 +1,278 @@
+
+Each component has their own `index.js` and `locators.js` files stored in `carbon/cypress/locators/component-name` unless the locators are common to more than one component/element/story, in which case they are contained in `carbon/cypress/locators`.
+
+###### How to locate elements:
+Always use unique selectors to locate elements with the order below:
+1.	`[data-component="unique-tag"]` for component and `[data-element="unique-tag"]` for component's element
+2.	`#id`
+3.	`.class-name`
+
+###### How to name locators:
+*  Component on preview - `componentPreview`
+*  Knobs - `componentKnobs`
+
+An example of an `index.js` file:
+```
+import { BUTTON_SUBTEXT_PREVIEW, BUTTON_DATA_COMPONENT_PREVIEW } from './locators';
+
+// component preview locators
+export const buttonSubtextPreview = () => cy.iFrame(BUTTON_SUBTEXT_PREVIEW);
+export const buttonDataComponent = () => cy.iFrame(BUTTON_DATA_COMPONENT_PREVIEW);
+
+// component preview locators in no IFrame
+export const buttonDataComponentNoIFrame = () => cy.get(BUTTON_DATA_COMPONENT_PREVIEW);
+```
+
+An example of a corresponding `locators.js` file:
+
+```
+// component preview locators
+export const BUTTON_SUBTEXT_PREVIEW = '[data-element="subtext"]';
+export const BUTTON_DATA_COMPONENT_PREVIEW = '[data-component="button"]';
+```
+
+Other support files include the `helper.js` file which contains functions such as that used to prepare the URL of the component page and the control of draggable components.
+
+
+##### Test Results
+Depending on how Cypress was run, you will find test results in either the Cypress UI, the console or in GitHub Actions linked from the build checks in GitHub. For further information see the [Getting Started Guide](https://github.com/Sage/carbon/blob/master/cypress/README.md).
+
+##### Applitools
+###### Integration with Cypress
+Instructions for adding Applitools to an existing Cypress project can be found [here](https://applitools.com/tutorials/cypress.html). Applitools is integrated with Cypress by means of various configuration and settings in the `cypress/support/index.js` file, `applitools.config.js` file, and with additional feature files and corresponding step definitions of those components/stories where we want to do visual testing. When the relevant tests are run from Cypress, snapshots will be taken where specified in the feature files. We currently run visual tests in Chrome browser only.
+
+An example of a feature file for a visual testing step follows. The `Then` step is where Applitools will take a snapshot:
+
+```
+Feature: Button all buttons component
+  I want to check that all examples of Button component render correctly per theme
+
+  @positive
+  @applitools
+  Scenario Outline: Check that all buttons render correctly with theme set to <theme>
```

Comment:
there is no such test in repo :(

Reply:
I know, it was just meant to be an example of how one could be written. I'll change it to an actual example to avoid confusion.

Reply:
I used the `Button component - primary type` spec as an example here.